pkg/cmd/run.go

```diff
@@ -162,3 +204,24 @@ func controlplaneFunc() func(r *envoy.Runtime) {
 		return func(r *envoy.Runtime) {}
 	}
 }
+
+// Function verifies template parameters passed in getenvoy command line.
+// If verification is successful, it returns a string which must be added to
+// Envoy command line to invoke a proper config.
+func processTemplateArgs(flavor string, args []string, templateParams map[string]string, runtime *envoy.Runtime) (string, error) {
+	for _, envoyParam := range args {
+		if strings.HasPrefix(envoyParam, "--config") {
+			return "", fmt.Errorf("--templateArg and %s cannot be specified at the same time", envoyParam)
+		}
+	}
+	config, err := flavors.CreateConfig(flavor, templateParams)
+	if err != nil {
+		return "", err
+	}
+	// Save config in getenvoy directory
+	path, err := runtime.SaveConfig(flavor, config)
```

Comment:
I think we should generate permanent config files and keep overwriting them. The reason is that a user may bootstrap basic Postgres (or any other flavor) config, copy it out, modify to their purpose and run getenvoy with custom config.

Reply:
My concern is that such a config file will be constantly overwritten.

So, although the file is there, a user cannot really rely on it. We might even need to warn a user to make a copy of this file if he doesn't want to lose it.

Reply:
Another question, we save the file but don't tell the user about it.

What is the value of that file then ? (from a user perspective)

Reply:
Another thing to consider, there must be a way for a user to learn more about configuration templates supported by `getenvoy`, their arguments, usage examples, etc.

Let's say something like `getenvoy template list | show <template_name> | render <template_name>`

Then a user could generate a config file by running `getenvoy template render <template_name> --arg endpoint=host1:5432`.

And we don't need to store configs.

Reply:
Hmm, I have to rethink it. Templates are derived from image flavor, like postgres:nightly, standard:1.15, istio:1.15. Based on the prefix, the corresponding template is found. Right now there is only postgres, but other ones like mysql can be added. Image type is also chosen based on prefix and it is guaranteed that image is going to work with a config template. If we allow specifying image type and template separately, it could lead to conflicts.

Reply:
Captured it in new issue: https://github.com/tetratelabs/getenvoy/issues/97. Closing it for now.